### PR TITLE
Add note about IDistributedCache

### LINKED ISSUE #27813: Cache storage: Explain why IDistributedCache is not recommended in docs
In Cache Storage section, please explain why IDistributedCache is not recommended in docs. 

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: cb423d54-ee93-6dfd-87bb-ed79dff8b7b6
* Version Independent ID: cb423d54-ee93-6dfd-87bb-ed79dff8b7b6
* Content: [Output caching middleware in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/performance/caching/output?view=aspnetcore-7.0#cache-storage)
* Content Source: [aspnetcore/performance/caching/output.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/performance/caching/output.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @tdykstra
* Microsoft Alias: **riande**

---
[Associated WorkItem - 46092](https://dev.azure.com/msft-skilling/Content/_workitems/edit/46092)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #27813

### CODE REVIEW

File: aspnetcore/performance/caching/output.md [hunk 1]
```diff
@@ -157,7 +157,7 @@ The following properties of <xref:Microsoft.AspNetCore.OutputCaching.OutputCache
 
 ## Cache storage
 
-<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching.
+<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, and that makes tagging brittle.
```
Comment (sebastienros) [inline aspnetcore/performance/caching/output.md:160]:
```suggestion
<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, and that makes tagging brittle. We recommend to create custom <xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> implementations using direct dependencies on the underlying storage mechanism, like Redis.
```

Comment (sebastienros) [reply aspnetcore/performance/caching/output.md:160]: I added more thoughts to it, feel free to use it as you want.

Comment (Rick-Anderson) [inline aspnetcore/performance/caching/output.md:160]: > that makes tagging **_brittle_**.

_brittle_ probably won't MT

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/performance/caching/output.md (hunk 1 above)
Comment (tdykstra) [reply aspnetcore/performance/caching/output.md:160]:
```suggestion
<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, which are required for tagging.
```

Comment (Rick-Anderson) [reply aspnetcore/performance/caching/output.md]: awesome

File: aspnetcore/performance/caching/output.md [hunk 2]
```diff
@@ -157,7 +157,7 @@ The following properties of <xref:Microsoft.AspNetCore.OutputCaching.OutputCache
 
 ## Cache storage
 
-<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching.
+<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, which are required for tagging.
```
Comment (tdykstra) [inline aspnetcore/performance/caching/output.md:160]:
```suggestion
<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, which are required for tagging.  We recommend that you create custom <xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> implementations by using direct dependencies on the underlying storage mechanism, such as Redis.
```

File: aspnetcore/performance/caching/output.md [hunk 3]
```diff
@@ -157,7 +157,7 @@ The following properties of <xref:Microsoft.AspNetCore.OutputCaching.OutputCache
 
 ## Cache storage
 
-<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching.
+<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, which are required for tagging.  We recommend that you create custom <xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> implementations by using direct dependencies on the underlying storage mechanism, such as Redis.
```
Comment (tdykstra) [inline aspnetcore/performance/caching/output.md:160]: removing extra space
```suggestion
<xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> is used for storage. By default it's used with <xref:System.Runtime.Caching.MemoryCache>. We don't recommend <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> for use with output caching. `IDistributedCache` doesn't have atomic features, which are required for tagging. We recommend that you create custom <xref:Microsoft.AspNetCore.OutputCaching.IOutputCacheStore> implementations by using direct dependencies on the underlying storage mechanism, such as Redis.
```

**Pull request merged**